R/rectangle.R

```diff
@@ -167,104 +222,461 @@ hoist <- function(.data, .col, ..., .remove = TRUE, .simplify = TRUE, .ptype = l
 check_pluckers <- function(...) {
   pluckers <- list2(...)
 
-  is_string <- vapply(pluckers, function(x) is.character(x) && length(x) == 1, logical(1))
+  is_string <- map_lgl(pluckers, ~is.character(.x) && length(.x) == 1)
   auto_name <- names2(pluckers) == "" & is_string
 
   if (any(auto_name)) {
     names(pluckers)[auto_name] <- unlist(pluckers[auto_name])
   }
 
-  if (!is_named(pluckers)) {
-    stop("All elements of `...` must be named", call. = FALSE)
+  if (length(pluckers) > 0 && !is_named(pluckers)) {
+    abort("All elements of `...` must be named.")
   }
 
   if (vec_duplicate_any(names(pluckers))) {
-    abort("The names of `...` must be unique")
+    abort("The names of `...` must be unique.")
   }
 
+  # Standardize all pluckers to lists for splicing into `pluck()`
+  # and for easier handling in `strike()`
+  is_not_list <- !map_lgl(pluckers, vec_is_list)
+  pluckers[is_not_list] <- map(pluckers[is_not_list], vec_chop)
```

Comment:
It is generally easier to transform a mix of plucker types like: `hoist(df, col, x = "x", y = c(1, 2), z = list("foo", "bar"))` into their equivalent list type, like `hoist(df, col, x = list("x"), y = list(1, 2), z = list("foo", "bar"))`